src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,624 @@
+  type = "MA",
+  test = "welch",
+  xlab = "Log-ratio abundance",
+  ylab = "Difference",
+  cutoff = 0.05)
+
+# Dispersion plot (right)
+aldex.plot(aldex_out,
+  type = "MW",
+  test = "welch",
+  xlab = "Dispersion",
+  ylab = "Difference",
+  cutoff = 0.05)
+
+# Reset plotting layout
+par(mfrow = c(1, 1))
+
+# Then do some further evaluation on the results
+aldex_out %>%
+  rownames_to_column(var = "Genus") %>%
+  # here we choose the wilcoxon output rather than t-test output
+  filter(wi.eBH <= 0.05)  %>%
+  dplyr::select(Genus, we.eBH, wi.eBH, effect, overlap) %>%
+  knitr::kable()
+```
+
+## Confounder Testing 
+
+Test the associated metadata variables for potential confounding influence. As many biological and technical factors beyond the primary phenotype of interest can influence microbiome composition, simple association studies may suffer confounding by other variables, which can lead to spurious results. Associations are visualized either as barplot or Q-Q plot, depending on the type of metadata. Additionally, it evaluates associations among metadata variables using conditional entropy and associations with the label using generalized linear models, producing a correlation heatmap and appropriate quantitative barplots, respectively.
+
+Here we use the `ANCOMBC` package to test for cofounders.
+
+1. Compute and store the library size in `colData`.
+
+```{r}
+genus_tse$library_size <- colSums(assay(genus_tse, "counts"))
+```
+
+
+2. The `ancombc2_out` method performs the differential abundance analysis whilst accounting for multiple covariates and potential confounders. 
+
+```{r}
+# Perform the analysis 
+ancombc2_out <- ancombc2(genus_tse,
+                         assay.type = "counts",
+                         fix_formula = "sample_geographic.location..country.and.or.sea.region. + library_size",
+                         p_adj_method = "fdr",
+                         lib_cut = 0,
+                         group = "sample_geographic.location..country.and.or.sea.region.", 
+                         struc_zero = TRUE, 
+                         neg_lb = TRUE,
+                         alpha = 0.05,
+                         global = TRUE)
+```
+
+## Generation of a predictive model: Machine learning workflow
+
+We'll now use the `mikropml` library to construct a machine learning meta-analysis models on the basis of microbial markers. The steps to follow include functions for data normalization, splitting the data into cross-validation folds, training the model, and making predictions based on cross-validation instances and the trained models. Such a model can be built on one dataset and then be applied on another, similarly processed holdout dataset. This might be of interest when comparing data from two different studies testing the same condition.
+
+1. Normalize the data using CLR transformation.
+
+```{r}
+filtered_tse <- transformAssay(filtered_tse, assay.type = "counts", method = "clr", MARGIN = "samples", pseudocount = 1)
```

Comment:
Usually we avoid filtering in general and only do it in specific analyses, for instance before DAA to reduce multiple testing.

Is it necessary to use filtered data here? The ML workflow might benefit from access to the full count data (or transformations derived from it).